modules/label/modules/barcode/commerce_pos_label_barcode.module

```diff
@@ -71,19 +74,28 @@ function commerce_pos_label_barcode_generate($product, $format) {
   $generator = new BarcodeGeneratorPNG();
   $upc_field = variable_get('commerce_pos_label_barcode_upc_field_' . $product->type, FALSE);
 
-  if (!empty($upc_field) && $field_upc = field_get_items('commerce_product', $product, $upc_field)) {
-    try {
-      $barcode_settings = $format['barcode'];
-
-      $barcode =  $generator->getBarcode($field_upc[0]['value'], $barcode_settings['type'], $barcode_settings['widthFactor'], $barcode_settings['totalHeight'], $barcode_settings['color']);
-      return $barcode;
+  if(!empty($upc_field)) {
+    if($upc_field == 'sku') {
```

Comment:
Would it be nicer to use an `EntityMetadataWrapper` for `$product` so we can do `$product_wrapper->{$upc_field}->value()` without caring what the name of it is?